storage/time-series/influxdb/retriever/app/influx/DataQuery.js

```diff
@@ -0,0 +1,272 @@
+  fluxInteger,
+  fluxDateTime,
+  fluxString,
+} = require('@influxdata/influxdb-client');
+const util = require('util');
+const { Logger } = require('@dojot/microservice-sdk');
+
+const logger = new Logger('influxdb-retriever:influx/DataQuery');
+
+/**
+ * This class handle with query data in a specific bucket.
+ * @class
+ */
+class DataQuery {
+  /**
+   *
+   * @param {String} url   Url to access influxdb
+   * @param {String} token  A token with write permission in all orgs
+   * @param {String} defaultBucket  Bucket Name for all data write
+   *
+   */
+  constructor(url, token, defaultBucket) {
+    logger.debug('constructor:');
+    logger.debug(`constructor: url=${url}`);
+    logger.debug(`constructor: token=${token}`);
+    logger.debug(`constructor: defaultBucket=${defaultBucket}`);
+
+    this.influxDB = new InfluxDB({ url, token });
+    this.defaultBucket = defaultBucket;
+    // prefix adds to all fields to be written
+    this.prefixFields = 'dojot.';
+    this.prefixFieldsSize = (this.prefixFields).length;
+  }
+
+  /**
+ * Fetch data for a given field considering the time
+ * slot and paging filter for a default bucket and an given org.
+ *
+ * @param {String} org Organization Name
+ * @param {String} measurement Measurement Name
+ * @param {object} filters Filters for query
+ * @param {string} filters.dateFrom=1970-01-01T00:00:00.000Z
+ * @param {string} filter.dateTo=(current time)
+ * @param {object} page Paginate information for query
+ * @param {number} page.limit=256
+ * @param {number} page.page=1
+ * @param {{String='desc','asc'}} order=desc Defines whether the order by **time** should be
+ *                                            ascending (asc) or descending (desc)
+ *
+ * @returns {Promise.<{result: [{ ts: timeIsoData, attrs: [label:string, value:any }],
+ *                            totalItems: number}| error>}
+ *                            A promise that returns a result e a totalItems
+ */
+  async queryByMeasurement(org, measurement, filters = {}, page = {}, order = 'desc') {
+    try {
+      logger.debug('queryByMeasurement:');
+      logger.debug(`queryByMeasurement: org=${org}`);
+      logger.debug(`queryByMeasurement: measurement=${measurement}`);
+      logger.debug(`queryByMeasurement: filters=${util.inspect(filters)}`);
+      logger.debug(`queryByMeasurement: page=${util.inspect(page)}`);
+
+      const {
+        start, stop, limit, offset,
+      } = DataQuery.commonQueryParams(page, filters);
+
+      const orderExp = DataQuery.commonQueryOrderExpression(order);
+
+      const fluxQuery = flux`from(bucket:${fluxString(this.defaultBucket)})
+      |> range(start: ${start} , stop: ${stop})
+      |> filter(fn: (r) => r._measurement == ${fluxString(measurement)})
+      |> drop(columns: ["_start", "_stop", "_measurement"])
+      |> pivot(rowKey:["_time"], columnKey: ["_field"], valueColumn: "_value")
+      ${fluxExpression(orderExp)}
+      |> limit(n: ${limit} , offset: ${offset})`;
```

Comment:
We proposed a paging approach using limit and offset, which is not wrong; but it would be more efficient a strategy based on time displacements, i.e,  manipulating the dataFrom and dataTo in order to reduce the data set in each iteration.